pilot/pkg/bootstrap/server.go

```diff
@@ -492,6 +493,18 @@ func (s *Server) initKubeClient(args *PilotArgs) error {
 		if err != nil {
 			return fmt.Errorf("failed creating kube client: %v", err)
 		}
+
+		memberRollName := args.RegistryOptions.KubeOptions.MemberRollName
+		if memberRollName != "" {
+			err := s.kubeClient.AddMemberRoll(args.Namespace, memberRollName)
+			if err != nil {
+				return fmt.Errorf("failed creating member roll: %v", err)
+			}
+		} else {
+			// No MemberRoll configured, set namespaces based on args.
+			namespaces := strings.Split(args.RegistryOptions.KubeOptions.WatchedNamespaces, ",")
```

Comment:
this might be problematic, as it's going to end up as empty slice instead of a slice with an empty string (which would mean all namespaces)

Reply:
Oops. True.

Reply:
except of course if we merge my pr: https://github.com/maistra/xns-informer/pull/4

Reply:
Fixed.